clang/lib/CodeGen/MicrosoftCXXABI.cpp

```diff
@@ -1131,13 +1132,18 @@ static bool isTrivialForMSVC(const CXXRecordDecl *RD, QualType Ty,
     return false;
   if (RD->hasNonTrivialCopyAssignment())
     return false;
+  if (RD->needsImplicitCopyAssignment() && !RD->hasSimpleCopyAssignment())
```

Comment:
I didn't find a `defaultedCopyAssignmentIsDeleted()` method on `CXXRecordDecl` however `hasSimpleCopyAssignment()` does what we want here when combined with `needsImplicitCopyAssignment()`.

Let me know if there is a better check to capture implicitly declared but deleted copy assignment operators on the record.

Reply:
If I'm understanding correctly, this is basically split into two parts:

- If we implicitly declared the copy-assignment, just query isDeleted() on it directly.
- If we didn't implicitly declare the copy-assignment, check whether it would have been deleted.  (We can't use this in cases where the copy-assignment was declared because the information isn't stored in the CXXRecordDecl.)

This is kind of complicated, but I can't think of a simpler way to do it.  That said, it probably deserves a comment to explain the logic.

Reply:
Correct. The two main cases we need to cover are the following. 
I'll make a comment to explain the logic.

```
struct Test
{
    Test& operator=(const Test&) = delete;
    int x;
};
```
In this case `DefaultedCopyAssignmentIsDeleted` is `false` on the `CXXRecordDecl`. So we need to query `isDeleted` on the copy assignment operator `CXXMethodDecl`.

```
struct Test
{
    int& x;
};
```
In this case we have no `CXXMethodDecl` for the copy assignment operator so we need to check if `DefaultedCopyAssignmentIsDeleted` is `true` on the `CXXRecordDecl`.